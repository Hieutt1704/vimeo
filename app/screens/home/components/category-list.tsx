import React from "react"
import { FlatList, View } from "react-native"
import { Category } from "../../../models"
import homeScreenStyles from "../home-screen.styles"
import { CategoryItem } from "./category-item"

const keyExtractor = (item: Category) => item.key

const itemSeparatorComponent = () => <View style={homeScreenStyles.categorySeparator} />

interface CategoryListProps {
  testID?: string
  categories: Category[]
}

export const CategoryList = ({ categories, testID }: CategoryListProps) => {
  const renderItem = ({ item }) => {
    return <CategoryItem category={item} />
  }

  return (
    <FlatList
      testID={testID}
      data={categories}
      numColumns={2}
      keyExtractor={keyExtractor}
      contentContainerStyle={homeScreenStyles.flatList}
      renderItem={renderItem}
      ItemSeparatorComponent={itemSeparatorComponent}
    />
  )
}
