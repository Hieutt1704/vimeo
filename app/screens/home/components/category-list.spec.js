import { render, screen } from "@testing-library/react-native"
import React from "react"
import { CategoryList } from "./category-list"

test("category-list no data", () => {
  render(<CategoryList categories={[]} />)
  expect(screen.toJSON()).toMatchSnapshot()
})

test("category-list have data", () => {
  render(
    <CategoryList
      categories={[
        {
          key: "abcdef",
          id: "favourite",
          name: "favourite",
          lastUpdate: "022-08-20T18:51:49+00:00",
          image:
            "https://i.vimeocdn.com/grab?s=https%3A%2F%2Ff.vimeocdn.com%2Fimages_v6%2Fcategories%2Firis_icon_sports_64.png%3F11f6d29ab6a020eb65c580a92c1a836c25cf81c4%3Fv%3D2&r=pad&w=20&h=20&f=png",
        },
        {
          key: "abcdefasdasd",
          id: "favourite",
          name: "favourite",
          lastUpdate: "022-08-20T18:51:49+00:00",
          image:
            "https://i.vimeocdn.com/grab?s=https%3A%2F%2Ff.vimeocdn.com%2Fimages_v6%2Fcategories%2Firis_icon_sports_64.png%3F11f6d29ab6a020eb65c580a92c1a836c25cf81c4%3Fv%3D2&r=pad&w=20&h=20&f=png",
        },
      ]}
    />,
  )
  expect(screen.toJSON()).toMatchSnapshot()
})
