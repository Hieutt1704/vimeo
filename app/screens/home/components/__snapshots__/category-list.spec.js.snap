// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category-list have data 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingVertical": 8,
    }
  }
  data={
    Array [
      Object {
        "id": "favourite",
        "image": "https://i.vimeocdn.com/grab?s=https%3A%2F%2Ff.vimeocdn.com%2Fimages_v6%2Fcategories%2Firis_icon_sports_64.png%3F11f6d29ab6a020eb65c580a92c1a836c25cf81c4%3Fv%3D2&r=pad&w=20&h=20&f=png",
        "key": "abcdef",
        "lastUpdate": "022-08-20T18:51:49+00:00",
        "name": "favourite",
      },
      Object {
        "id": "favourite",
        "image": "https://i.vimeocdn.com/grab?s=https%3A%2F%2Ff.vimeocdn.com%2Fimages_v6%2Fcategories%2Firis_icon_sports_64.png%3F11f6d29ab6a020eb65c580a92c1a836c25cf81c4%3Fv%3D2&r=pad&w=20&h=20&f=png",
        "key": "abcdefasdasd",
        "lastUpdate": "022-08-20T18:51:49+00:00",
        "name": "favourite",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 8,
            }
          }
        >
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="cover"
            source={
              Array [
                Object {
                  "uri": "https://i.vimeocdn.com/video/1486234694-2789256b3e16a01f78abaebcc3eeb1ef872b63bc8bfda834c8c4cc6b11fb578b-d",
                },
              ]
            }
            style={
              Object {
                "height": 363,
                "overflow": "hidden",
                "width": 363,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00000090",
                "height": 363,
                "justifyContent": "center",
                "position": "absolute",
                "width": 363,
              }
            }
          >
            <RCTImageView
              internal_analyticTag={null}
              resizeMode="cover"
              source={
                Array [
                  Object {
                    "uri": "https://i.vimeocdn.com/grab?s=https%3A%2F%2Ff.vimeocdn.com%2Fimages_v6%2Fcategories%2Firis_icon_sports_64.png%3F11f6d29ab6a020eb65c580a92c1a836c25cf81c4%3Fv%3D2&r=pad&w=20&h=20&f=png",
                  },
                ]
              }
              style={
                Object {
                  "height": 24,
                  "overflow": "hidden",
                  "tintColor": "#ffffff",
                  "width": 24,
                }
              }
              tintColor="#ffffff"
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginTop": 8,
                  },
                ]
              }
            >
              favourite
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginLeft": 8,
            }
          }
        >
          <RCTImageView
            internal_analyticTag={null}
            resizeMode="cover"
            source={
              Array [
                Object {
                  "uri": "https://i.vimeocdn.com/video/1486234694-2789256b3e16a01f78abaebcc3eeb1ef872b63bc8bfda834c8c4cc6b11fb578b-d",
                },
              ]
            }
            style={
              Object {
                "height": 363,
                "overflow": "hidden",
                "width": 363,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00000090",
                "height": 363,
                "justifyContent": "center",
                "position": "absolute",
                "width": 363,
              }
            }
          >
            <RCTImageView
              internal_analyticTag={null}
              resizeMode="cover"
              source={
                Array [
                  Object {
                    "uri": "https://i.vimeocdn.com/grab?s=https%3A%2F%2Ff.vimeocdn.com%2Fimages_v6%2Fcategories%2Firis_icon_sports_64.png%3F11f6d29ab6a020eb65c580a92c1a836c25cf81c4%3Fv%3D2&r=pad&w=20&h=20&f=png",
                  },
                ]
              }
              style={
                Object {
                  "height": 24,
                  "overflow": "hidden",
                  "tintColor": "#ffffff",
                  "width": 24,
                }
              }
              tintColor="#ffffff"
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginTop": 8,
                  },
                ]
              }
            >
              favourite
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`category-list no data 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingVertical": 8,
    }
  }
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
